Issue #1219 restructure troubleshooting topic

thatdocslady opened the pull request:
Fixes issue #1219 

Per agreement on the issue, the troubleshooting docs are in their own category with each sub-category in its own topic file. Also, inside the driver plug-in topic, there is an additional breakdown per driver.

This PR was tested locally and topic map is updated.

hferentschik commented:
@thatdocslady Your refactoring left some dangling links:

    11:41:45 - build/getting-started/installing.html
    11:41:45   *  internally linking to ../using/troubleshooting.html#machine-doesnt-exist, which     does not exist (line 237)
    11:41:45      <a href="../using/troubleshooting.html#machine-doesnt-exist">Error: machine does     not exist</a>
    11:41:45   *  internally linking to ../using/troubleshooting.html, which does not exist (line     238)
    11:41:45      <a href="../using/troubleshooting.html">Troubleshooting</a>
    11:41:45   *  trying to find hash of ../using/troubleshooting.html#machine-doesnt-exist, but /    home/docs/build/using/troubleshooting.html does not exist (line 237)
    11:41:45      <a href="../using/troubleshooting.html#machine-doesnt-exist">Error: machine does     not exist</a>
    11:41:45 - build/getting-started/quickstart.html
    11:41:45   *  internally linking to ../using/    troubleshooting.html#minshift-startup-check-failed, which does not exist (line 195)
    11:41:45      <a href="../using/troubleshooting.html#minshift-startup-check-failed">    troubleshooting</a>
    11:41:45   *  trying to find hash of ../using/    troubleshooting.html#minshift-startup-check-failed, but /home/docs/build/using/    troubleshooting.html does not exist (line 195)
    11:41:45      <a href="../using/troubleshooting.html#minshift-startup-check-failed">    troubleshooting</a>
    11:41:45 - build/using/index.html
    11:41:45   *  internally linking to ../using/troubleshooting.html, which does not exist (line     141)
    11:41:45      <a href="../using/troubleshooting.html">Troubleshooting</a>
    11:41:45 htmlproofer 3.7.2 | Error:  HTML-Proofer found 6 failures!

`make link_check_docs` is your friend here.

hferentschik commented:
@thatdocslady let's also move Troubleshooting above Contributing

hferentschik commented on docs/source/troubleshooting/troubleshooting-driver-plugins.adoc:
```diff
@@ -0,0 +1,216 @@
+:icons:
+:toc: macro
+:toc-title:
+:toclevels: 1
```
You need at least doc level of 2 here, otherwise the items will be impossible to find

hferentschik commented on docs/source/_topic_map.yml:
```diff
@@ -59,6 +57,17 @@ Topics:
         File: writing-docs
       - Name: Releasing Minishift
         File: releasing
+  - Name: Troubleshooting Minishift
```
I would be happy with just "Troubleshooting"

hferentschik commented on docs/source/_topic_map.yml:
```diff
@@ -59,6 +57,17 @@ Topics:
+    Topics:
+      - Name: Overview
+        File: index
+      - Name: Troubleshooting General
```
I move this last and I'd prefer the term 'Misc' or 'Miscellaneous'. IMO general does not quite have the right meaning.

thatdocslady commented:
@hferentschik I fixed the broken links and tested everything again. All your feedback points are now implemented, except that I think we should keep "Troubleshooting Minishift" for consistency (we qualify whether it's minishift/openshift in all the other categories).

hferentschik commented:
>  I fixed the broken links and tested everything again. 

thanks

> All your feedback points are now implemented,

Ok

> except that I think we should keep "Troubleshooting Minishift" for consistency (we qualify whether it's minishift/openshift in all the other categories).

fair enough

hferentschik commented:
Merged, thanks.
